[TDX-2081]: update developer registration and SSO docs (#3994)
Kong/docs.konghq.com: app/konnect/dev-portal/customization/index.md

content_type: how-to
---

> The {{site.konnect_short_name}} Dev Portal can be customized in the following ways:

Reviewer: ```suggestion
You can customize the {{site.konnect_short_name}} Dev Portal in the following ways:
```

   {:.note}
   > **Note:** DNS propagation can take a few hours. If after a few hours you can't access the Dev Portal from the custom URL, contact your domain registrar.

## Single Sign-On
{:.badge .enterprise}

> The Dev Portal can be configured to use Single Sign-On (SSO) by a {{site.konnect_short_name}} Admin via the Identity settings in the {{site.konnect_short_name}} Admin interface.

Reviewer: ```suggestion
A {{site.konnect_short_name}} Admin can configure Single Sign-On (SSO) for the Dev Portal via the Identity settings in the {{site.konnect_short_name}} Admin interface.
```

1. In {{site.konnect_short_name}}, open {% konnect_icon dev-portal %} **Dev Portal**, then click **Settings**.

2. Open the **Identity** tab.

   {:.note}
      > **Note:** Copy the callback URL and enter it in your OIDC Provider

> 3. Enter the full domain, including subdomain and protocol, into the  **Provider URL** field (also known as **Issuer**). e.g. `https://accounts.google.com` for Google IdP

Reviewer: ```suggestion
3. Enter the full domain, including the subdomain and protocol, into the  **Provider URL** field (also known as **Issuer**). For example,  `https://accounts.google.com` for Google IdP.
```

4. Enter the unique identifier provided by the IdP into the **Client ID** field.

5. Enter the secret used to verify ownership of your IdP client into the **Client Secret** field.

### Configuration Requirements

> A form of authentication must always be configured. Built-in and SSO can be used individually or in combination. Each state results in a different user experience represented in this table:

Reviewer: ```suggestion
You must always have a form of authentication configured. Built-in and SSO can be used individually or in combination. Each state results in a different user experience, as represented in the following table:
```

| State | SSO registration | SSO sign in | Built-in registration| Built-in sign in | Information|
| --- | :-----------: |  ----------- | :-----------: | ----------- |

> | **SSO: Enabled**<br>**Built-in: Disabled** | ✅ | ✅ | ❌ | ❌ | - Developer will be able to register and log in with SSO.<br><br>- Developers registered with Built-in will be forced to use SSO.<br><br>- The Built-in authentication window will not be displayed from the Dev portal. 

Reviewer: ```suggestion
| **SSO: Enabled**<br>**Built-in: Disabled** | ✅ | ✅ | ❌ | ❌ | - Developers can register and log in with SSO.<br><br>- Developers registered with Built-in are forced to use SSO.<br><br>- The Built-in authentication window does not display from the Dev Portal. 
```

> | **SSO: Disabled**<br>**Built-in: Enabled** | ❌ | ❌ | ✅  |✅  | - Developers will be able to register and log in with a username and password.| 

Reviewer: ```suggestion
| **SSO: Disabled**<br>**Built-in: Enabled** | ❌ | ❌ | ✅  |✅  | - Developers can register and log in with a username and password.| 
```

> | **SSO: Enabled**<br>**Built-in: Enabled** | ✅  | ✅ | ✅  | ✅  | - Developers will be able to register and log in with either form of authentication.|

Reviewer: ```suggestion
| **SSO: Enabled**<br>**Built-in: Enabled** | ✅  | ✅ | ✅  | ✅  | - Developers can register and log in with either form of authentication.|
```

### OIDC Details

> * If a user account associated to a {{site.konnect_short_name}} developer is removed from the IdP, the {{site.konnect_short_name}} developer account will continue to exist. It must be deleted by a {{site.konnect_short_name}} admin from the {{site.konnect_short_name}} dashboard.

Reviewer: ```suggestion
* If a user account associated with a {{site.konnect_short_name}} developer is removed from the IdP, the {{site.konnect_short_name}} developer account is not deleted. A {{site.konnect_short_name}} Admin must delete it from the {{site.konnect_short_name}} dashboard.
```
"...associated with a...." might just be a personal preference, so feel free to ignore that. 

Also, you used "Konnect Admin" before, so I just capped "admin" because I couldn't find a rule in the style guide about it.

> * If a {{site.konnect_short_name}} developer associated with an IdP user is deleted, the same IdP user can re-authenticate with the developer portal and a new {{site.konnect_short_name}} developer account will be created. To persistently revoke access for developers authenticating through your IdP, you must remove the ability for that user to authenticate with the IdP.

Reviewer: ```suggestion
* If a {{site.konnect_short_name}} developer associated with an IdP user is deleted, the same IdP user can re-authenticate with the Dev Portal and a new {{site.konnect_short_name}} developer account is created. To persistently revoke access for developers authenticating through your IdP, you must remove the ability for that user to authenticate with the IdP.
```

   {:.note}
   > **Note:** DNS propagation can take a few hours. If after a few hours you can't access the Dev Portal from the custom URL, contact your domain registrar.

<!-- markdownlint-disable-next-line -->
## Single Sign-On
{:.badge .enterprise}

Single Sign-On (SSO) can be configured in the Dev Portal by those with Admin roles via the Identity settings in the {{site.konnect_short_name}} Admin UI.

1. In {{site.konnect_short_name}}, open {% konnect_icon dev-portal %} **Dev Portal**, then click **Settings**.

2. Open the **Identity** tab.

3. Enter the full domain, including subdomain and protocol, into the  **Provider URL** field (also known as **Issuer**). e.g. `https://accounts.google.com` for Google IdP

4. Enter the unique identifier provided by the IdP into the **Client ID** field.

> 5. Enter the secret used to verify ownership of your IdP client into the **Client Secret** field.

Reviewer 2: Maybe extra step like "Make sure you provide Callback URL to your IdP"?
